stablehlo/tests/CheckOps.cpp

```diff
@@ -75,6 +81,48 @@ llvm::Error evalExpectEqOp(const Tensor &lhs, const Tensor &rhs) {
   return llvm::Error::success();
 }
 
+// Fetch a previously serialized filepath given a `probeId` and a `probeDir` for
+// a specified `iteration` value from an `index.csv` metadata file. If no data
+// is found, returns an error.
+static llvm::ErrorOr<std::string> getSerializedTensorPath(StringRef probeId,
+                                                          StringRef probeDir,
+                                                          uint32_t iteration) {
+  llvm::SmallString<128> instrumentationMetadataFile(probeDir);
```

Comment:
This may already be handled by the current implementation, but make sure this errors if:
1. probeDir is empty
2. probeDir / metadataFile doesn't exist (probe was never called)

Let's also make sure we have a test for this. Can help with negative testing if needed.

Reply:
Should we lock this down in a separate cc test? I tried to lock down via the LIT `expected-error` mechanism, but it appears it's thrown as a fatal assertion and it didn't seem to work well with that mechanism.

Reply:
Ah, you're right. If a fallback fails, we still report fatal error: https://github.com/openxla/stablehlo/blob/371f5145c31a0210d9d400192d9c8d6d579431ae/stablehlo/reference/Ops.cpp#L815

We can skip the negative tests for now. Instead of a separate CC test for the sake of testing report_fatal, I say we address this separately from this PR. @ghpvnist, what are your thoughts - I think a better solution longer term would be to refactor the interpreter to report errors more gracefully than report_fatal, so we can test them in lit tests. In the rest of the codebase report_fatal is used for things that should never happen.

Reply:
I agree. There's already a ticket #1196 to improve error reporting for better debugging experience on tensors. As you mentioned, so far the interpreter errors were unexpected. With additional users to the interpreter, I think it makes sense to refactor to allow lit based testing.